Philips.CodeAnalysis.MaintainabilityAnalyzers/Readability/PreferInterpolatedStringAnalyzer.cs

```diff
@@ -0,0 +1,190 @@
+
+		private void OnInvocation(OperationAnalysisContext operationContext)
+		{
+			var invocation = (IInvocationOperation)operationContext.Operation;
+
+			if (!IsStringFormatMethod(invocation.TargetMethod))
+			{
+				return;
+			}
+
+			ConversionResult result = CanConvertToInterpolatedString(invocation);
+			if (result == ConversionResult.None)
+			{
+				return;
+			}
+
+			Location location = invocation.Syntax.GetLocation();
+			if (result.HasFlag(ConversionResult.IsUnnecessary))
+			{
+				operationContext.ReportDiagnostic(Diagnostic.Create(UnnecessaryRule, location));
+			}
+			else
+			{
+				operationContext.ReportDiagnostic(Diagnostic.Create(Rule, location));
+			}
+		}
+
+		private bool IsStringFormatMethod(IMethodSymbol targetMethod)
+		{
+			return targetMethod.Name == "Format" &&
+				   targetMethod.ContainingType != null &&
+				   targetMethod.ContainingType.SpecialType == SpecialType.System_String;
+		}
+
+		private ConversionResult CanConvertToInterpolatedString(IInvocationOperation invocation)
+		{
+			if (invocation.Arguments.Length < 1)
+			{
+				return ConversionResult.None;
+			}
+
+			IOperation formatStringArgument = invocation.Arguments[0].Value;
+
+			if (formatStringArgument.Kind != OperationKind.Literal ||
+				formatStringArgument.Type?.SpecialType != SpecialType.System_String)
+			{
+				return ConversionResult.None;
+			}
+
+			var formatString = (string)formatStringArgument.ConstantValue.Value;
+
+			// Parse the format string to find actual placeholders, ignoring escaped braces
+			var placeholderCount = CountFormatPlaceholders(formatString);
+
+			if (placeholderCount == 0)
+			{
+				// Only flag as unnecessary if there are no braces at all
+				// If there are escaped braces, string.Format is needed to produce literal braces
+				var hasAnyBraces = formatString.Contains("{") || formatString.Contains("}");
+				if (!hasAnyBraces)
+				{
+					return ConversionResult.CanConvert | ConversionResult.IsUnnecessary;
+				}
+
+				// Has escaped braces but no placeholders - don't suggest conversion
+				return ConversionResult.None;
+			}
+
+			return invocation.Arguments.Length > 1 ? ConversionResult.CanConvert : ConversionResult.None;
+		}
+
+		private int CountFormatPlaceholders(string formatString)
+		{
+			var placeholderCount = 0;
+			var i = 0;
+
+			while (i < formatString.Length)
+			{
+				if (formatString[i] == '{')
+				{
+					if (i + 1 < formatString.Length && formatString[i + 1] == '{')
+					{
+						// Escaped brace {{, skip both characters
+						i += 2;
+						continue;
+					}
+
+					// Look for closing brace
+					var j = i + 1;
+					while (j < formatString.Length && formatString[j] != '}')
+					{
+						j++;
+					}
+
+					if (j < formatString.Length && formatString[j] == '}')
+					{
+						// Found valid placeholder, check if it's a simple numeric placeholder
+						var content = formatString.Substring(i + 1, j - i - 1);
+
+						// Handle format specifiers like {0:N2} by splitting on ':'
```

Comment:
We should also handle alignment options (denoted with a comma). https://learn.microsoft.com/en-us/previous-versions/visualstudio/visual-studio-2008/txafckwd(v=vs.90)#alignment-component  In such cases, it's OK to just abort the operation, though currently we'll try to TryParse "0,3" for example. (Unit test)